commit da82c216b22797fa5edc4dd91246be629e205532 (refs/changes/08/4108/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-06-01T13:48:08+02:00 (1 year, 4 months ago)
    
    joyent/node-cueball#138 want method to know if Agent is stopped

diff --git a/docs/api.adoc b/docs/api.adoc
index 968ca8b..ae99c37 100644
--- a/docs/api.adoc
+++ b/docs/api.adoc
@@ -136,6 +136,11 @@ held by the pool). The `stop()` method should be used on an Agent if your
 program does not plan to use it again and wishes to be able to exit node
 without calling `process.exit()`.
 
+### `#isStopped()`
+
+Boolean, is this Agent stopped?. Useful to prevent exceptions being thrown
+when calling `stop()` more than once.
+
 ### `#getPool([host])`
 
 Returns the connection pool this agent has created for the input host. If no
diff --git a/lib/agent.js b/lib/agent.js
index fbae298..3edd3fc 100644
--- a/lib/agent.js
+++ b/lib/agent.js
@@ -491,6 +491,10 @@ CueBallAgent.prototype.createConnection = function (options, connectListener) {
 	throw (new Error('UNIX domain sockets not supported'));
 };
 
+CueBallAgent.prototype.isStopped = function () {
+	return this.cba_stopped;
+};
+
 function CueBallHttpAgent(options) {
 	options.protocol = 'http';
 	if (options.defaultPort === undefined)
